File: reuse_java/src/main/java/com/kmmania/AthleteImpl.java
package com.kmmania;

/**
 * The AthleteImpl class implements the {@link AthleteInterface} and represents an athlete.
 *
 * @author Dac Linh Giao NGUYEN
 * @version 0.1
 * @see AthleteInterface
 */
public class AthleteImpl implements AthleteInterface {
    private final String speciality;

    /**
     * Constructs an AthleteImpl object with the specified specialty.
     *
     * @param speciality the specialty of the athlete
     */
    public AthleteImpl(String speciality) {
        this.speciality = speciality;
    }

    /**
     * Returns a string representation of the athlete's specialty.
     *
     * @return a string representing the athlete's specialty
     */
    @Override
    public String mySpeciality() {
        return "My speciality is " + this.speciality + ".";
    }
}

File: reuse_kotlin/src/main/kotlin/AthleteImpl.kt
/*
 * Classes reusability project.
 * Dac Linh Giao NGUYEN
 * License MIT
 *
 * Classes reusability in Kotlin.
 * AthleteImpl.kt
 */
package com.kmmania

/**
 * Implementation of the AthleteInterface representing an athlete.
 *
 * @property speciality The specialty of the athlete.
 */
class AthleteImpl(private val speciality: String) : AthleteInterface {

    /**
     * Retrieves the speciality of the athlete.
     *
     * @return A string representing the speciality of the athlete.
     */
    override fun mySpeciality(): String = "My speciality is $speciality."
}